### PR TITLE
[2.13.x] Changed PKI certificate validation

### PR DESCRIPTION
**Pull request by blen-desta**
#### What does this PR do?
PKI authentication certificates will now be validated against the truststore.

#### Who is reviewing it? 
@bakejeyner @SmithJosh 

#### Select relevant component teams: 
@codice/security 

#### Ask 2 committers to review/merge the PR and tag them here.
@garrettfreibott
@stustison 

#### How should this be tested?
- Create a CA (`rootCA`)
- Create a certificate using the `rootCA` (`cert`)
- Add the `rootCA` to DDF's **keystore** (**NOT the truststore**)
- Use the `cert` to login to DDF -> this should fail

- **Remove** the `rootCA` from the DDF's **keystore**
- **Add** the `rootCA` to DDF's **truststore**
- Use the `cert` to login to DDF -> this should pass

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [x] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by blen-desta** [comment]
build now

**Comment by blen-desta** [comment]
build now

**Comment by blen-desta** [comment]
build now

**Comment by bakejeyner** [comment]
Hero ✔️

**Comment by blen-desta** [comment]
build now

File: platform/security/handler/security-handler-api/src/main/java/org/codice/ddf/security/handler/api/PKIAuthenticationTokenFactory.java [hunk 1]
```diff
@@ -29,17 +37,23 @@
...
   /** Initializes Merlin crypto object. */
   public void init() {
-    try {
-      merlin =
-          new Merlin(
-              PropertiesLoader.loadProperties(signaturePropertiesPath),
-              PKIAuthenticationTokenFactory.class.getClassLoader(),
-              null);
-    } catch (WSSecurityException | IOException e) {
+    // NOTE: THE TRUSTSTORE SHOULD BE USED FOR CERTIFICATE VALIDATION!!!!
```
Comment (jlcsmith) [inline platform/security/handler/security-handler-api/src/main/java/org/codice/ddf/security/handler/api/PKIAuthenticationTokenFactory.java]: ❓  this appears to be a duplicate code block to the one above.  Can we move it into a method that both classes can use instead?

**Pull request merged**